x-pack/plugins/data_enhanced/public/search/sessions_mgmt/components/status.tsx

```diff
@@ -0,0 +1,195 @@
+        defaultMessage: 'Complete',
+      });
+    case STATUS.ERROR:
+      return i18n.translate('xpack.data.mgmt.searchSessions.status.label.error', {
+        defaultMessage: 'Error',
+      });
+    default:
+      // eslint-disable-next-line no-console
+      console.error(`Unknown status ${statusType}`);
+      return statusType;
+  }
+};
+
+interface StatusIndicatorProps {
+  now?: string;
+  session: UISession;
+  timezone: string;
+}
+
+// Get the fields needed to show each status type
+// can throw errors around date conversions
+const getStatusAttributes = ({
+  now,
+  session,
+  timezone,
+}: StatusIndicatorProps): StatusAttributes | null => {
+  switch (session.status) {
+    case STATUS.IN_PROGRESS:
+      try {
+        const createdDate = dateString(session.created, timezone);
+        return {
+          textColor: 'default',
+          icon: <EuiLoadingSpinner />,
+          label: <TableText>{getStatusText(session.status)}</TableText>,
+          toolTipContent: i18n.translate(
+            'xpack.data.mgmt.searchSessions.status.message.createdOn',
+            {
+              defaultMessage: 'Started on {createdDate}',
+              values: { createdDate },
+            }
+          ),
+        };
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error(err);
+        throw new Error(`Could not instantiate a createdDate object from: ${session.created}`);
+      }
+
+    case STATUS.EXPIRED:
+      try {
+        const expiredOnDate = dateString(session.expires!, timezone);
+        const toolTipContent = i18n.translate(
+          'xpack.data.mgmt.searchSessions.status.message.expiredOn',
+          {
+            defaultMessage: 'Expired on {expiredOnDate}',
+            values: { expiredOnDate },
+          }
+        );
+
+        return {
+          icon: <EuiIcon color="#9AA" type="clock" />,
+          label: <TableText>{getStatusText(session.status)}</TableText>,
+          toolTipContent,
+        };
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error(err);
+        throw new Error(`Could not instantiate a expiration Date object from: ${session.expires}`);
+      }
+
+    case STATUS.CANCELLED:
+      return {
+        icon: <EuiIcon color="#9AA" type="crossInACircleFilled" />,
+        label: <TableText>{getStatusText(session.status)}</TableText>,
+        toolTipContent: i18n.translate('xpack.data.mgmt.searchSessions.status.message.cancelled', {
+          defaultMessage: 'Cancelled by user',
+        }),
+      };
+
+    case STATUS.ERROR:
+      return {
+        textColor: 'danger',
+        icon: <EuiIcon color="danger" type="crossInACircleFilled" />,
+        label: <TableText>{getStatusText(session.status)}</TableText>,
+        toolTipContent: i18n.translate('xpack.data.mgmt.searchSessions.status.message.error', {
+          defaultMessage: 'Error: {error}',
+          values: { error: (session as any).error || 'unknown' },
```

Comment:
Where does the error field come from?

Reply:
It currently isn't part of the mapping of the session attributes, but it should be added.

Probably needs to be populated during session monitoring polling and updating?